### ISSUE TITLE
Can only submit once?

### ISSUE THREAD

**Issue by djfm** [erikras/redux-form#1641]
# Problem Description

I'm doing async submit validation as described [in the doc](http://redux-form.com/6.0.1/examples/submitValidation/).

If my `onSubmit` rejects with an error, I cannot submit the form again.

It also does it in the official doc!

I input "John":

![image](https://cloud.githubusercontent.com/assets/1460499/18033742/cd6c01c0-6d2c-11e6-8b78-788d3714eff3.png)

Then add the correct password ("redux-form") and get the same error:

![image](https://cloud.githubusercontent.com/assets/1460499/18033746/e6907870-6d2c-11e6-9470-1cfcf564a696.png)

Whereas if I input "John" "redux-form" after refreshing the page:

![image](https://cloud.githubusercontent.com/assets/1460499/18033750/09f9cd8e-6d2d-11e6-81af-c0683bdf99f7.png)

Is this expected? That's kinda weird...

# Workaround

So I ended up doing:

```javascript
reduxForm({
    form: 'login',
    destroyOnUnmount: false,
    validate: () => ({}),
})
```

because `handleSubmit` does not run submit if the form is not `valid`:

```javascript
// lib/handleSubmit.js
if (valid) {
    var doSubmit = function doSubmit() {
      var result = void 0;
      try {
        result = submit(values, dispatch);
      } catch (submitError) {
        var error = submitError instanceof _SubmissionError2.default ? submitError.errors : undefined;
        setSubmitFailed.apply(undefined, _toConsumableArray(fields));
        if (onSubmitFail) {
          onSubmitFail(error, dispatch);
        }
        return error;
      }
  // etc.
}
```
My interpretation of the issue is that the doc is wrong when it says that the default value of the `validate` function is `() => ({})`:

![image](https://cloud.githubusercontent.com/assets/1460499/18034320/30193d3a-6d3a-11e6-8f36-db50e9bb5667.png)

I really like this library and I'd be glad to submit a fix if you're interested and my interpretation of the issue is correct!

In my opinion just having the correct default value for `validate` should be enough.

**Comment by JakeLin1024** [comment]
because handleSubmit does not run submit if the form is not valid:

I ran into a use case which is related to this.

I have two fields named start date and end date(validation rule: end date should be later than start date)
if I have a end date(2016/8/28) that is before start date(2016/8/29), we show a error msg below the end date.
But the **problem** is:
If I correct the start date(2016/8/27, this will match the rule) I can't submit the form and the error below end date still persist because the end date is still invalid and it leads to a invalid form.

To fix this my workaround is submit the form after calling ```stopSubmit(form)``` action.
But I kinda don't prefer doing this though.

**Comment by ortex** [comment]
It is correct worked in early version. http://redux-form.com/6.0.0-alpha.5/examples/submitValidation/

**Comment by cruciux** [comment]
@ortex does handleSubmit only allow the submit function to get called if the form gets changed after failing the first time? If this is the case then it doesn't account for the form failing to submit due to network issues or something on the server etc.

**Comment by ortex** [comment]
Now in function `createIsValid` check `error`
```
const error = getIn(formState, `${form}.error`)
if (error) {
  return false
}
```

As far as I understand, `error` never not deleted.

**Comment by erikras** [comment]
The reason this broke was that previous versions were incorrectly storing the `_error` in the submit and async validation sections of the store, but `v6.0.1` rectified this. It just missed clearing the global error on `CHANGE`. Thanks for the fix, @ortex. I'll comment here again when this is published.

**Comment by erikras** [comment]
Fix published in [`v6.0.2`](https://github.com/erikras/redux-form/releases/tag/v6.0.2).

**Comment by djfm** [comment]
I confirm resolution, thanks!
